src/library/coverart.h

```diff
@@ -85,7 +85,36 @@ class CoverInfo : public CoverInfoRelative {
     CoverInfo(CoverInfo&&) = default;
     CoverInfo& operator=(CoverInfo&&) = default;
 
-    QImage loadImage(
+    struct LoadedImage final {
+        enum class Result {
+            Ok,
+            NoImage,
+            ErrorMetadataWithEmptyTrackLocation,
+            ErrorRelativeFilePathWithEmptyTrackLocation,
+            ErrorFilePathDoesNotExist,
+            ErrorLoadingFailed, // if the image file is not readable or the format is not supported
+            ErrorUnknown,
+        };
+
+        LoadedImage()
+                : result(Result::ErrorUnknown) {
```

Comment:
This is correct, because it will never happen. See the DEBUG_ASSERT in the loadImage() function.

I will hide the default constructor, because LoadedImage should only be created by the loadImage() function.